### PR TITLE
os: fix getenvb, scandir

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

**Comment by srittau** [comment]
Now conflicts with master.

**Comment by hauntsaninja** [comment]
Sorry about that, fixed!

File: stdlib/3/os/__init__.pyi [hunk 1]
```diff
@@ -361,7 +361,10 @@ if sys.platform != 'win32':
 def getenv(key: Text) -> Optional[str]: ...
 @overload
 def getenv(key: Text, default: _T) -> Union[str, _T]: ...
-def getenvb(key: bytes, default: bytes = ...) -> bytes: ...
+@overload
+def getenvb(key: bytes) -> Optional[bytes]: ...
+@overload
+def getenvb(key: bytes, default: _T = ...) -> Union[bytes, _T]: ...
```
Comment (srittau) [inline stdlib/3/os/__init__.pyi]: `default` is explicitly documented to be `bytes`, but using another type seems to work fine.

Comment (hauntsaninja) [reply stdlib/3/os/__init__.pyi]: At the very least, `default` is by default None... I can change this to:
```
@overload
 def getenvb(key: bytes, default: None = ..) -> Optional[bytes]: ...
 @overload
 def getenvb(key: bytes, default: bytes) -> bytes: ...
```
Might help surface errors sooner if someone passes a str as default.

Comment (srittau) [reply stdlib/3/os/__init__.pyi]: It's fine, since we prefer actual behavior over documentation.

**Pull request merged**